File: exam-slammr-webapp/src/main/java/uk/co/claritysoftware/exam/slammr/webapp/web/config/SocialProviders.java
package uk.co.claritysoftware.exam.slammr.webapp.web.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

import lombok.Value;

/**
 * A simple bean encapsulating which Social Providers are configured
 */
@Value
@Component
public class SocialProviders {

	private final boolean facebookConfigured;

	private final boolean twitterConfigured;

	private final boolean linkedinConfigured;

	@Autowired
	public SocialProviders(ApplicationContext applicationContext) {
		facebookConfigured = applicationContext.containsBean("facebook");
		twitterConfigured = applicationContext.containsBean("twitter");
		linkedinConfigured = applicationContext.containsBean("linkedin");
	}
}

File: exam-slammr-webapp/src/main/kotlin/uk/co/claritysoftware/exam/slammr/webapp/web/config/SocialProviders.kt
package uk.co.claritysoftware.exam.slammr.webapp.web.config

import lombok.Value
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationContext
import org.springframework.stereotype.Component

/**
 * A simple bean encapsulating which Social Providers are configured
 */
@Value
@Component
class SocialProviders @Autowired
constructor(applicationContext: ApplicationContext) {

	val isFacebookConfigured: Boolean = applicationContext.containsBean("facebook")

	val isTwitterConfigured: Boolean = applicationContext.containsBean("twitter")

	val isLinkedinConfigured: Boolean = applicationContext.containsBean("linkedin")
}
